src/enum_net.cpp

```diff
@@ -587,12 +587,18 @@ int _System __libsocket_sysctl(int* mib, u_int namelen, void *oldp, size_t *oldl
 #if TORRENT_USE_IFADDRS && !defined TORRENT_BUILD_SIMULATOR
 	bool iface_from_ifaddrs(ifaddrs *ifa, ip_interface &rv)
 	{
+		// some android ifaddrs implementations are buggy
+		if (ifa->ifa_addr == nullptr) return false;
+
```

Comment:
why not have the three blocks the same way?

```cpp
if (ifa->ifa_addr != nullptr)
{
        rv.interface_address = sockaddr_to_address(ifa->ifa_addr);
}
```

(this works well when I test it)